test/Microsoft.ML.Tests/Scenarios/IrisPlantClassificationTests.cs

```diff
@@ -3,33 +3,45 @@
 // See the LICENSE file in the project root for more information.
 
 using Microsoft.ML.Data;
-using Microsoft.ML.Legacy.Models;
-using Microsoft.ML.Legacy.Trainers;
-using Microsoft.ML.Legacy.Transforms;
+using Microsoft.ML.RunTests;
 using Xunit;
-using TextLoader = Microsoft.ML.Legacy.Data.TextLoader;
 
 namespace Microsoft.ML.Scenarios
 {
-#pragma warning disable 612, 618
     public partial class ScenariosTests
     {
         [Fact]
-        public void TrainAndPredictIrisModelTest()
+        public void New_TrainAndPredictIrisModelTest()
         {
-            string dataPath = GetDataPath("iris.txt");
-
-            var pipeline = new Legacy.LearningPipeline(seed: 1, conc: 1);
-
-            pipeline.Add(new TextLoader(dataPath).CreateFrom<IrisData>(useHeader: false));
-            pipeline.Add(new ColumnConcatenator(outputColumn: "Features",
-                "SepalLength", "SepalWidth", "PetalLength", "PetalWidth"));
-
-            pipeline.Add(new StochasticDualCoordinateAscentClassifier());
-
-            Legacy.PredictionModel<IrisData, IrisPrediction> model = pipeline.Train<IrisData, IrisPrediction>();
-
-            IrisPrediction prediction = model.Predict(new IrisData()
+            var mlContext = new MLContext(seed: 1, conc: 1);
+
+            var reader = mlContext.Data.CreateTextReader(columns: new[]
+                {
+                    new TextLoader.Column("Label", DataKind.R4, 0),
+                    new TextLoader.Column("SepalLength", DataKind.R4, 1),
+                    new TextLoader.Column("SepalWidth", DataKind.R4, 2),
+                    new TextLoader.Column("PetalLength", DataKind.R4, 3),
+                    new TextLoader.Column("PetalWidth", DataKind.R4, 4)
+                }
+            );
+
+            var pipe = mlContext.Transforms.Concatenate("Features", "SepalLength", "SepalWidth", "PetalLength", "PetalWidth")
+                            .Append(mlContext.Transforms.Normalize("Features"))
+                            .AppendCacheCheckpoint(mlContext)
```

Comment:
Why are we adding normalizer transform? just curious.

Reply:
I think in the new API  the paradigm is to be explicit  about whether are using  Normalizers  in the pipeline .  

(In the old Legacy API normalizers used to be auto-inserted based on the algorithm)

Hence decided to be explicit about normalizers.  Please see #1604  which talks about  removing 'smarts' (i.e.  no auto-normalization, no auto-caching and no auto-calibration) in the new API


---
In reply to: [245111952](https://github.com/dotnet/machinelearning/pull/2008#discussion_r245111952) [](ancestors = 245111952)